ast/src/parsed/mod.rs

```diff
@@ -633,6 +635,51 @@ pub enum BinaryOperator {
     Greater,
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub enum BinaryOperatorAssociativity {
+    Left,
+    Right,
+}
+
+impl BinaryOperator {
+    pub fn precedence(&self) -> ExpressionPrecedence {
+        use BinaryOperator::*;
+        match self {
+            // Unary - * ! & &mut
+            Pow => 2,
+            // * / %
+            Mul | Div | Mod => 3,
+            // + -
+            Add | Sub => 4,
+            // << >>
+            ShiftLeft | ShiftRight => 5,
+            // &
+            BinaryAnd => 6,
+            // ^
+            BinaryXor => 7,
+            // |
+            BinaryOr => 8,
+            // == != < > <= >=
+            Equal | NotEqual | Less | Greater | LessEqual | GreaterEqual => 9,
+            // &&
+            LogicalAnd => 10,
+            // ||
+            LogicalOr => 11,
+            // .. ..=
+            // ??
+            // = += -= *= /= %= &= |= ^= <<= >>=
+            Identity => 12,
```

Comment:
This should be 9

Reply:
@chriseth Do you mean it must be 9 and others starting from 9 right now be shifted by one?!

Reply:
According to https://doc.rust-lang.org/reference/expressions.html#expression-precedence
It's at the end of the list. But as I look at it now, I missed the point that it's right associative.

Reply:
What matters is not the precedence in the rust documentation but the precedence as implemented by our grammar: https://github.com/powdr-labs/powdr/blob/main/parser/src/powdr.lalrpop#L469

While our grammar mostly aligns with that of rust, the identity is one point where it diverges, because Identity is not an assignment but another kind of operator.